lib/bricks-net-url.js

```diff
@@ -0,0 +1,335 @@
+			
+			cppArguments.assert('QueryParameters', [
+				[
+					cppArguments.assertion('string', 'const std::string&', 'key'),
+					function (key) {
+						retval = (typeof _this.parameters_[key] === 'string');
+					}
+				]
+			], arguments);
+			
+			return retval;
+		};
+		QueryParameters.prototype.get = function () {
+			var _this = this;
+			var retval = '';
+			
+			cppArguments.assert('QueryParameters', [
+				[
+					cppArguments.assertion('string', 'const std::string&', 'key'),
+					cppArguments.assertion('string', 'const std::string&', 'default_value', cppArguments.ASSERTION_MODE_OPTIONAL),
+					function (key, default_value) {
+						if (typeof default_value === 'undefined') {
+							default_value = '';
+						}
+						
+						retval = (_this.has(key) ? _this.parameters_[key] : default_value);
+					}
+				]
+			], arguments);
+			
+			return retval;
+		};
+	}
+	function URLParametersExtractor_ComposeParameters() {
+		var composed_parameters = "";
+		for (var i = 0, ic = _this.parameters_vector.length; i < ic; ++i) {
+			composed_parameters += "?&".charAt(i > 0 ? 1 : 0);
+			composed_parameters += encodeURIComponent(_this.parameters_vector[i].first) + '=' +
+				encodeURIComponent(_this.parameters_vector[i].second);
+		}
+		if (_this.fragment) {
+			composed_parameters += "#" + _this.fragment;
+		}
+		return composed_parameters;
+	}
+	
+	
+	cppArguments.assert('URL', [
+		[],
+		[
+			cppArguments.assertion('string', 'const std::string&', 'url'),
+			cppArguments.assertion('string', 'const std::string&', 'previous_scheme', cppArguments.ASSERTION_MODE_OPTIONAL),
+			cppArguments.assertion('string', 'const std::string&', 'previous_host', cppArguments.ASSERTION_MODE_OPTIONAL),
+			cppArguments.assertion('int', 'const int', 'previous_port', cppArguments.ASSERTION_MODE_OPTIONAL),
+			function (url, previous_scheme, previous_host, previous_port) {
+				if (typeof previous_scheme === 'undefined') {
+					previous_scheme = kDefaultScheme;
+				}
+				if (typeof previous_host === 'undefined') {
+					previous_host = "";
+				}
+				if (typeof previous_port === 'undefined') {
+					previous_port = 0;
+				}
+				
+				URLParametersExtractor(url);
+				URLWithoutParametersParser(
+					_this.url_without_parameters,
+					previous_scheme, previous_host, previous_port
+				);
+			}
+		],
+		[
+			cppArguments.assertion('string', 'const std::string&', 'url'),
+			cppArguments.assertion(function (value) {
+				return (value instanceof URL);
+			}, 'const URLWithoutParametersParser&', 'previous'),
+			function (url, previous) {
+				URLParametersExtractor(url);
+				URLWithoutParametersParser(
+					_this.url_without_parameters,
+					previous
+				);
+			}
+		]
+	], arguments);
+	
+	this.ComposeParameters = function () {
+		return URLParametersExtractor_ComposeParameters();
+	};
+	this.ComposeURL = function () {
+		return URLWithoutParametersParser_ComposeURL() + URLParametersExtractor_ComposeParameters();
+	};
+}
+
+exports.URL = URL;
```

Comment:
In general, it's covered by the tests. There are a couple of cases which were not covered, will fix as soon as I get to reviewing overall code coverage.